src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java

```diff
@@ -0,0 +1,282 @@
+                .remove(entities));
+    }
+
+
+    @Override
+    public Mono<Void> delete(Publisher<? extends T> entityStream) {
+        Assert.notNull(entityStream, "The given publisher of entities must not be null!");
+        return Mono.empty(Flux.from(entityStream)
+                .flatMap(entity -> operations.remove(entity)));
+    }
+
+
+    @Override
+    public Mono<Long> count() {
+        final ResolvedView resolvedView = determineView();
+        ViewQuery query = ViewQuery.from(resolvedView.getDesignDocument(), resolvedView.getViewName());
+        query.reduce(true);
+        query.stale(operations.getDefaultConsistency().viewConsistency());
+
+        return operations.queryView(query)
+                .flatMap(ReactiveViewResult::rows)
+                .single()
+                .map(row -> Long.valueOf(row.value().toString()));
+    }
+
+    @Override
+    public Mono<Void> deleteAll() {
+        final ResolvedView resolvedView = determineView();
+        ViewQuery query = ViewQuery.from(resolvedView.getDesignDocument(), resolvedView.getViewName());
+        query.reduce(false);
+        query.stale(operations.getDefaultConsistency().viewConsistency());
+
+
+        return Mono.empty(operations.queryView(query)
+                .map(ReactiveViewResult::rows)
+                .flatMap(row -> {
+                    ReactiveViewRow reactiveViewRow = (ReactiveViewRow) row;
+                    return operations.remove(reactiveViewRow.id())
+                            .otherwise(throwable -> {
+                                if (throwable instanceof DocumentDoesNotExistException) {
+                                    return Mono.empty();
+                                }
+                                return Mono.error(throwable);
+                            });
+                }));
+
+
+    }
+
+    /**
+     * Returns the information for the underlying template.
+     *
+     * @return the underlying entity information.
+     */
+    protected CouchbaseEntityInformation<T, String> getEntityInformation() {
+        return entityInformation;
+    }
+
+    /**
+     * Resolve a View based upon:
+     * <p/>
+     * 1. Any @View annotation that is present
+     * 2. If none are found, default designDocument to be the entity name (lowercase) and viewName to be "all".
+     *
+     * @return ResolvedView containing the designDocument and viewName.
+     */
+    private ResolvedView determineView() {
+        String designDocument = StringUtils.uncapitalize(entityInformation.getJavaType().getSimpleName());
+        String viewName = "all";
+
+        final View view = viewMetadataProvider.getView();
+
+        if (view != null) {
+            designDocument = view.designDocument();
+            viewName = view.viewName();
+        }
+
+        return new ResolvedView(designDocument, viewName);
+    }
+
+    @Override
+    public ReactiveCouchbaseOperations getCouchbaseOperations(){
+        return operations;
```

Comment:
That's the only bit which would not be converted when using reactive type adoption from the repository layer.